x-pack/test/security_solution_api_integration/test_suites/detections_response/utils/rules/prebuilt_rules/install_fleet_package_by_url.ts

```diff
@@ -14,6 +14,7 @@ import { refreshSavedObjectIndices } from '../../refresh_index';
 
 const MAX_RETRIES = 2;
 const ATTEMPT_TIMEOUT = 120000;
+const TOTAL_TIMEOUT = ATTEMPT_TIMEOUT * (1 + MAX_RETRIES);
```

Comment:
@xcrzx Addressed.

> Is ATTEMPT_TIMEOUT enforced anywhere? If I understand correctly, the first attempt will continue running even if it exceeds the 2-minute timeout. This means only the total timeout of 6 minutes is effectively being enforced.

That's correct. The `ATTEMPT_TIMEOUT` is not enforced anywhere, and I was quite surprised by that. I think the API of this `tryWithRetries` method is not very intuitive. It looks like previously we thought that its `timeout` parameter applies to each attempt separately. By doing `TOTAL_TIMEOUT = ATTEMPT_TIMEOUT * (1 + MAX_RETRIES)` I thought it would help realize that this is not the case.